### ISSUE TITLE
Provides the ability to configure DefaultPersistenceConfiguration in CacheConfigurationBuilder

### ISSUE THREAD

**Issue by linghengqian** [ehcache/ehcache3#2997]
- In https://www.ehcache.org/blog/2016/05/18/ehcache3_jsr107_spring.html , it provided for the initial cache of JavaConfig in one idea.
```java
  @Component
  public static class CachingSetup implements JCacheManagerCustomizer
  {
    @Override
    public void customize(CacheManager cacheManager)
    {
      cacheManager.createCache("people", new MutableConfiguration<>()  
        .setExpiryPolicyFactory(TouchedExpiryPolicy.factoryOf(new Duration(SECONDS, 10))) 
        .setStoreByValue(false)
        .setStatisticsEnabled(true));
    }
  }
```
- At https://www.ehcache.org/blog/2017/03/15/spontaneous-cache-creation.html , Stéphane Nicoll (https://twitter.com/snicoll) also recommends this approach.
```java
@Bean
public JCacheManagerCustomizer cacheManagerCustomizer() {
    return cm -> {
        Configuration<Object, Object> cacheConfiguration = createCacheConfiguration();
        cm.createCache("vets", cacheConfiguration);
    };
}

private Configuration<Object, Object> createCacheConfiguration() {
    return Eh107Configuration.fromEhcacheCacheConfiguration(CacheConfigurationBuilder
        .newCacheConfigurationBuilder(Object.class, Object.class, ResourcePoolsBuilder
            .newResourcePoolsBuilder().heap(100)));
}
```
- But according to the official `ehcache3-example`, there is a detailed process in https://github.com/ehcache/ehcache3-samples/blob/master/fullstack/src/main/java/org/ehcache/sample/config/CacheConfiguration.java . In the configuration of this case, the `DefaultPersistenceConfiguration` can be configured by a code snippet similar to the following. But this should not be handled well in `ehcache3-example`(points to https://github.com/ehcache/ehcache3-samples ) as it will result. `public class CacheConfig extends CachingConfigurerSupport` should be changed to `public class CacheConfig implements CachingConfigurer`. Refer to https://github.com/spring-projects/spring-framework/issues/27813
```java
package com.lingh.config;

@EnableCaching
@Configuration
public class CacheConfig extends CachingConfigurerSupport {
    private javax.cache.CacheManager getCacheManager(EhcacheCachingProvider provider, DefaultConfiguration configuration) {
        return provider.getCacheManager(provider.getDefaultURI(), configuration);
    }

    @Bean 
    @Override
    public org.springframework.cache.CacheManager cacheManager() {
        return new JCacheCacheManager(createInMemoryCacheManager());
    }

    private javax.cache.CacheManager createInMemoryCacheManager() {
        CacheConfiguration<Object, Object> cacheConfiguration = CacheConfigurationBuilder.newCacheConfigurationBuilder(
                        Object.class, Object.class,
                        ResourcePoolsBuilder.newResourcePoolsBuilder().heap(123L, EntryUnit.ENTRIES)
                                .disk(123L, MemoryUnit.MB, true))
                .withExpiry(ExpiryPolicyBuilder.timeToLiveExpiration(Duration.ofDays(123L)))
                .build();
        Map<String, CacheConfiguration<?, ?>> caches = new HashMap<>();
        caches.putAll(this.createCacheConfigurations(cacheConfiguration));
        EhcacheCachingProvider provider = this.getCachingProvider();
        DefaultConfiguration configuration = new DefaultConfiguration(caches, provider.getDefaultClassLoader(), new DefaultPersistenceConfiguration(new File("./cache-data")));
        return this.getCacheManager(provider, configuration);
    }

    private Map<String, CacheConfiguration<?, ?>> createCacheConfigurations(CacheConfiguration<Object, Object> cacheConfiguration) {
        Map<String, CacheConfiguration<?, ?>> caches = new HashMap<>();
        caches.put("test", cacheConfiguration);
        return caches;
    }

     private EhcacheCachingProvider getCachingProvider() {
        return (EhcacheCachingProvider) Caching.getCachingProvider(EhcacheCachingProvider.class.getName());
    }
}
```
- But if you use the `JCacheManagerCustomizer` recommended by Stéphane Nicoll in https://www.ehcache.org/blog/2017/03/15/spontaneous-cache-creation.html , It is logically impossible to configure `DefaultPersistenceConfiguration` at all now. Because according to https://www.ehcache.org/documentation/3.9/107.html#cachemanager-level-configuration , configuring `DefaultPersistenceConfiguration` must be at the `org.ehcache.jsr107.EhcacheCachingProvider` level of `EhCache 3` to process. Since the case used by `ehcache3-example` is actually `jhipster`, I tried to find their latest use case, and what `jhipster` consider best practice now has become `JCacheManagerCustomizer`. Refer to https://github.com/jhipster/jhipster-sample-app/blob/main/src/main/java/io/github/jhipster/sample/config/CacheConfiguration.java . Refer to https://github.com/spring-projects/spring-boot/issues/3822 .

- To sum up, I hope EhCache 3 can provide the ability to configure `DefaultPersistenceConfiguration` in `CacheConfigurationBuilder`. Of course, if there is something I overlooked, it would be better to understand. Because https://stackoverflow.com/questions/65127520/set-ehcaches-persistence-directory-programmatically-on-spring-boot also has such doubts. Of course using `XML` to fully configure `EhCache 3` can completely solve similar problems, but I'd much rather see the `JavaConfig` way.

- There is also a related issue related to this, that is, the documentation for integrating with Spring has not been relatively complete. See https://github.com/ehcache/ehcache3/issues/1910

**Comment by linghengqian** [comment]
Refer to https://github.com/spring-projects/spring-boot/issues/3822 .

**Comment by linghengqian** [comment]
> * But according to the official `ehcache3-example`, there is a detailed process in https://github.com/ehcache/ehcache3-samples/blob/master/fullstack/src/main/java/org/ehcache/sample/config/CacheConfiguration.java . In the configuration of this case, the `DefaultPersistenceConfiguration` can be configured by a code snippet similar to the following. But this should not be handled well in `ehcache3-example`(points to https://github.com/ehcache/ehcache3-samples ) as it will result in a prompt like `2022-02-28 23:47:41.849 INFO 16048 --- [ restartedMain] trationDelegate$BeanPostProcessorChecker : Bean 'cacheConfig' of type [com.lingh.config.CacheConfig$$EnhancerBySpringCGLIB$$bfc041fe] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)`
> 

- Well, this was an oversight of mine, `public class CacheConfig extends CachingConfigurerSupport` should be changed to `public class CacheConfig implements CachingConfigurer`, but the same `INFO` problem persists. Refer to https://github.com/spring-projects/spring-framework/issues/27813

**Comment by linghengqian** [comment]
As a result, persistent bytes-based caching to hard disk is no longer the trend. This issue will be closed, refer to https://github.com/ehcache/ehcache3/issues/3008.
